### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by m8ttyB**
This adds Docker support.
* local development using a docker env is supported by this pr (Dockerfile + new .env files)
* Included in the pr are docker compose files for local development as well as for use in aws
* Updated Readme
* A json file containing fixture data for initializing new development environments

Todo - not in this pr
* A Readme on how to run the docker image in AWS

### CODE REVIEW

**Comment by m8ttyB** [comment]
pr #348 should remedy the build error.

**Comment by rbillings** [comment]
It looks like the build still has an error - I'll be happy to review once that is cleared up.

File: README.md [hunk 1]
```diff
@@ -124,6 +134,75 @@ you don't have `pip` installed, you can install it with `easy_install pip`.
...
+* [docs.docker.com/linux](https://docs.docker.com/linux/)
+* [docs.docker.com/windows](https://docs.docker.com/windows/)
+* [docs.docker.com/mac](https://docs.docker.com/mac/)
+ 
+#### Linux and OSX ####
+Initial setup is taken care of by the `setup.sh` script in this repository. There is a timing issue that the script corrects; the database needs to up and running before the Django server tries to connect to it.
+
+The dev environment runs on [http://localhost:8000](http://localhost:8000).
+
+##### Initial setup and use: #####
```
Comment (bebef1987) [inline README.md]: add sudo pip install docker-compose

**Comment by m8ttyB** [comment]
I'll update my the branch and add @bebef1987's suggestion
/cc @karlht

**Comment by karlht** [comment]
Matt, do you want to re-base this?

File: README.md (hunk 1 above)
Comment (m8ttyB) [inline README.md]: For reference, @bebef1987 was able to get a local install of docker on Linux working with One and Done.

**Comment by karlht** [comment]
@rbillings Are you still having local errors?  The Travis build seems fine now.

File: README.md [hunk 2]
```diff
@@ -124,6 +134,80 @@ you don't have `pip` installed, you can install it with `easy_install pip`.
...
+Note, the documentation assumes your Docker installation defaults to [http://localhost](http://localhost).
+
+Documentation specific for your operating system can be found here:
+
+* [docs.docker.com/linux](https://docs.docker.com/linux/)
+* [docs.docker.com/windows](https://docs.docker.com/windows/)
+* [docs.docker.com/mac](https://docs.docker.com/mac/)
+ 
+#### Linux and OSX ####
+Initial setup is taken care of by the `setup.sh` script in this repository. There is a timing issue that the script corrects; the database needs to up and running before the Django server tries to connect to it.
```
Comment (stephendonner) [inline README.md]: Uber-nit: "database needs to be up" (missing "be" in the original)

File: README.md [hunk 3]
```diff
@@ -124,6 +134,80 @@ you don't have `pip` installed, you can install it with `easy_install pip`.
...
+
+##### Initial setup and use: #####
+
+   ```
+   # Create the database first to ensure it is configured correctly before
+   # Django attempts to connect to it.
+   docker-compose up -d db
+   docker-compose up -d web
+
+   # Run django migrations
```
Comment (stephendonner) [inline README.md]: Uber-uber nit: capitalize "Django" since it is elsewhere

**Comment by rbillings** [comment]
@karlht - still errors, but only for me.  Details: https://public.etherpad-mozilla.org/p/Nk3xaB20K0

oneanddone_web |         Is the server running on host "localhost" (::1) and accepting
oneanddone_web |         TCP/IP connections on port 5432?
oneanddone_web | could not connect to server: Connection refused
oneanddone_web |         Is the server running on host "localhost" (127.0.0.1) and accepting
oneanddone_web |         TCP/IP connections on port 5432?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Install node and npm
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends nodejs npm;
+
+# install node modules
+RUN npm install less yuglify
+# create a symbolic link for node
+RUN ln -s /usr/bin/nodejs /usr/bin/node
+
+# Create created the required env and settings files
```
Comment (stephendonner) [inline Dockerfile]: "Create created"?

File: README.md [hunk 5]
```diff
@@ -24,6 +24,16 @@ About the project:
 
 Development Setup
 -----------------
+
+
+Developement can take place by either setting up a local environment that uses a locally installed PostgresSQL server or using Docker containers.
```
Comment (stephendonner) [inline README.md]: "Development"

File: docker/setup.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# On initial setup of the images run this script.
+
+# create the database first to ensure it is configured correctly before
```
Comment (stephendonner) [inline docker/setup.sh]: Uber-nit: capitalize "Create" to following established style?

File: README.md (hunk 2 above)
Comment (m8ttyB) [inline README.md]: Good catch, thank you! Fixed.

File: README.md (hunk 3 above)
Comment (m8ttyB) [inline README.md]: Fixed.

File: Dockerfile (hunk 4 above)
Comment (m8ttyB) [inline Dockerfile]: 👍 fixed

File: README.md (hunk 5 above)
Comment (m8ttyB) [inline README.md]: 👍 fixed

File: docker/setup.sh (hunk 6 above)
Comment (m8ttyB) [inline docker/setup.sh]: 👍 fixed

**Comment by stephendonner** [comment]
Really nice work, @m8ttyB!  I've run this locally using Docker/Docker Compose, and it's working just fine, so r+ from my end!
screen shot 2016-06-30 at 1 52 53 pm

**Comment by rbillings** [comment]
Adding the .env file to dockerignore did the trick! R+

**Comment by karlht** [comment]
Looks good to me.  @rbillings, will you push the magic button, please?

**Pull request merged**